src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestTrigger.java

```diff
@@ -70,28 +70,35 @@ private boolean shouldBuild(MergeRequestEvent mrEvent, GitLabSCMSourceContext co
                         getPayload().getObjectAttributes().getIid());
             }
 
+            LOGGER.log(Level.FINEST,
+                    "shouldBuild for MR-{0} will be set for action {1} based on pipeline configuration.",
+                    new Object[] {
+                            getPayload().getObjectAttributes().getIid(),
+                            action
+                    });
+
             if (action.equals("open")) {
-                return context.alwaysBuildMROpen();
+                return shouldBuild && context.alwaysBuildMROpen();
```

Comment:
This seems repeating why not just return early if `shouldBuild` is false?

Reply:
@jetersen : you are right. Fixed